File: backend/src/InventoryManagement.Domain/Entities/Product.cs
namespace InventoryManagement.Domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string SKU { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public int ReorderLevel { get; set; }
    public bool IsActive { get; set; } = true;
    
    // Foreign keys
    public Guid CategoryId { get; set; }
    public Guid SupplierId { get; set; }
    
    // Navigation properties
    public virtual Category Category { get; set; } = null!;
    public virtual Supplier Supplier { get; set; } = null!;
    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public virtual ICollection<StockMovement> StockMovements { get; set; } = new List<StockMovement>();
    public virtual ICollection<LowStockAlert> LowStockAlerts { get; set; } = new List<LowStockAlert>();
}

File: src/types/product.ts
export interface Product {
  id: string;
  name: string;
  description: string;
  sku: string;
  price: number;
  stockQuantity: number;
  reorderLevel: number;
  categoryId: string;
  supplierId: string;
  categoryName: string;
  supplierName: string;
  // category?: Category;
  // supplier?: Supplier;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface Category {
  id: string;
  name: string;
  description: string;
  isActive: boolean;
  productCount?: number;
}

export interface Supplier {
  id: string;
  name: string;
  contactEmail: string;
  contactPhone: string;
  address: string;
  isActive: boolean;
  productCount?: number;
}

export interface CreateProductRequest {
  name: string;
  description: string;
  sku: string;
  price: number;
  stockQuantity: number;
  reorderLevel: number;
  categoryId: string;
  supplierId: string;
}

export interface UpdateProductRequest extends CreateProductRequest {
  id: string;
}

export interface StockUpdateRequest {
  productId: string;
  quantity: number;
  reason: string;
}